localstack/services/cloudformation/resource_provider.py

```diff
@@ -397,11 +398,13 @@ def deploy_loop(
         self,
         resource: dict,
         raw_payload: ResourceProviderPayload,
-        max_iterations: int = 30,
+        max_timeout: int = config.CFN_PER_RESOURCE_TIMEOUT,
         sleep_time: float = 5,
     ) -> ProgressEvent[Properties]:
         payload = copy.deepcopy(raw_payload)
 
+        max_iterations = ceil(max_timeout/sleep_time)
```

Comment:
We may want to ensure at least 2 iterations...

```suggestion
        max_iterations = max(ceil(max_timeout/sleep_time), 2)
```